include/boost/geometry/algorithms/detail/envelope/implementation.hpp

```diff
@@ -0,0 +1,104 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2007-2015 Barend Gehrels, Amsterdam, the Netherlands.
+// Copyright (c) 2008-2015 Bruno Lalande, Paris, France.
+// Copyright (c) 2009-2015 Mateusz Loskot, London, UK.
+
+// This file was modified by Oracle on 2015.
+// Modifications copyright (c) 2015, Oracle and/or its affiliates.
+
+// Contributed and/or modified by Menelaos Karavelas, on behalf of Oracle
+
+// Parts of Boost.Geometry are redesigned from Geodan's Geographic Library
+// (geolib/GGL), copyright (c) 1995-2010 Geodan, Amsterdam, the Netherlands.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_ALGORITHMS_DETAIL_ENVELOPE_IMPLEMENTATION_HPP
+#define BOOST_GEOMETRY_ALGORITHMS_DETAIL_ENVELOPE_IMPLEMENTATION_HPP
+
+#include <boost/geometry/core/exterior_ring.hpp>
+#include <boost/geometry/core/tags.hpp>
+
+#include <boost/geometry/algorithms/expand.hpp>
+
+#include <boost/geometry/algorithms/detail/envelope/box.hpp>
+#include <boost/geometry/algorithms/detail/envelope/linestring.hpp>
+#include <boost/geometry/algorithms/detail/envelope/multilinestring.hpp>
+#include <boost/geometry/algorithms/detail/envelope/multipoint.hpp>
+#include <boost/geometry/algorithms/detail/envelope/point.hpp>
+#include <boost/geometry/algorithms/detail/envelope/range.hpp>
+#include <boost/geometry/algorithms/detail/envelope/segment.hpp>
+
+#include <boost/geometry/algorithms/dispatch/envelope.hpp>
+
+
+namespace boost { namespace geometry
+{
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail { namespace envelope
+{
+
+
+struct exterior_ring_expand_policy
+{
+    template <typename Box, typename Geometry>
+    static inline void apply(Box& mbr, Geometry const& geometry)
+    {
+        Box ring_mbr;
+        envelope_range<>::apply(exterior_ring(geometry), ring_mbr);
+        geometry::expand(mbr, ring_mbr);
```

Comment:
Ok, I get it. For me personally it isn't important how the expansion is implemented. It uses only the exterior ring because this is how we generally implement things for Polygon but it might do something else. It'd still expand a boxy using an envelope of a Polygon.

There is also `envelope_polygon` which also internally analyses only the exterior ring of a Polygon and `envelope_range_expand_policy` which expands a box using an envelope of a range. Well, the last name can also be confusing because this policy works on a Range of Points, not Range of envelopes (so boxes?).